### ISSUE TITLE
iOS build fails

### ISSUE THREAD

**Issue by as-radek** [rollbar/rollbar-react-native#180]
There is an error when building iOS (min: 14)

`[...]/Pods/RollbarCommon/RollbarCommon/Sources/RollbarCommon/DTOs/RollbarDTO+Protected.m:258:33 No known class method for selector 'rollbar_dateFromString:'`

as well as 
`[...]]/Pods/RollbarCommon/RollbarCommon/Sources/RollbarCommon/DTOs/RollbarDTO+Protected.m:287:38 No visible @interface for 'NSDate' declares the selector 'rollbar_toString'`

Followed the docs, but maybe there are any additional steps I should take? Looks like library issue though

**Comment by rollbar-bborsits** [comment]
Hey @as-radek, Can share which Rollbar SDK version you use?

**Comment by as-radek** [comment]
I'm installing 
* `rollbar-react-native`: `1.0.0-beta.0`
* `rollbar`: 2.26.2`
* `@rollbar/react`: `0.11.2`

According to podspec in `rollbar-react-native` it require `RollbarNotifier` for `3.0.3`

**Comment by rollbar-bborsits** [comment]
The [latest Apple SDK](https://github.com/rollbar/rollbar-apple/releases/tag/3.2.0) resolved this issue, but the React Native SDK wasn't updated yet. I file an internal ticket to update the React Native SDK.

**Comment by guilherme-vp** [comment]
Hi, are there any updates or workarounds for this issue? I am not able to use the SDK as well.

**Comment by Stas-Buzunko** [comment]
any updates on this??

**Comment by rollbar-bborsits** [comment]
@guilherme-vp The bugfix is on the backlog, but it will take some time to get there. We're working on some other high priority items currently.

**Comment by vpetrykov** [comment]
Good afternoon everyone, here's an update on the current issue:

- the problem persists in rollbar-react-native version 1.0.0-beta.1.
- after reviewing a [comment by repository contributor @waltjones](https://github.com/rollbar/rollbar-react-native/issues/178#issuecomment-1736468225) on a related issue [here](https://github.com/rollbar/rollbar-react-native/issues/178), it appears this issue should not exist in the newest 1.0.0-beta.1 version.
- on my end, I followed the [migration guidelines from 0.x.x to 1.x.x](https://docs.rollbar.com/docs/migrating-to-v100) (thanks for the informative documentation 👍), cleared all caches, and reinstalled CocoaPods, but I still encountered the 'unknown selector' issue.

Could someone provide more insight into this?

**Comment by waltjones** [comment]
In my testing, the originally reported errors are caused by `duplicate interface definition for class 'RollbarDTO'`, and that was caused by an inconsistent cache.

Using the `--no-build-cache` option solved this issue, and the build was successful.
```
expo run:ios  --no-build-cache
```

**Comment by waltjones** [comment]
I've tested different permutations with and without either `use_frameworks!` and `use_modular_headers!`. In all cases, the build succeeded when using `use_frameworks!` and failed as described above without `use_frameworks!`.

The most challenging part of testing these variations was ensuring I had a clean pod cache. Configurations would seem to succeed or fail, only to find cached pods were still being used from a previous build.

There is a new release available which uses the latest rollbar-apple dependency. This is the version I tested. If you're seeing the above error while `use_frameworks!` is set, it might be worthwhile to try the new release.

https://github.com/rollbar/rollbar-react-native/releases/tag/v1.0.0-beta.2
